tuf/api/metadata.py

```diff
@@ -730,7 +730,7 @@ class BaseFile:
 
     @staticmethod
     def _verify_hashes(
-        data: Union[bytes, BinaryIO], expected_hashes: Dict[str, str]
+        data: Union[bytes, IO[bytes]], expected_hashes: Dict[str, str]
```

Comment:
`BinaryIO` appears to be a subclass of `IO[bytes]` with the addition of interfaces to support context managers (`with`). We are not actually using the passed `data` object as a context manager, so this looks fine.